### ISSUE TITLE
pbfgraphparser.cc tagHandlers require exception handling

### ISSUE THREAD

**Issue by mkandulavm** [valhalla/valhalla#4245]
Hi Valhalla Team,

in src/mjolnir/pbfgraphparser.cc,
the stio conversions should be checked for exceptions, since the osm data is not consistent on this. The exception comes when the tag_.second below is not a int.
 
```
tag_handlers_["layer"] = [this]() {
      try {
        auto layer = static_cast<int8_t>(std::stoi(tag_.second));
        way_.set_layer(layer);
      } catch (const std::invalid_argument& ia) {
        // The string is not a number, so handle the error here.
        // For example, you might want to log a warning or set a default value:
        way_.set_layer(0);  // Use the default value you want here
      }
    };
```

Above is the fix I used. Kindly check and suggest.

**Comment by nilsnolde** [comment]
Yeah that's true, it's also not discarded in the graph.lua. I'd handle it by skipping the tag if it's not an int, no way to make sense of what it should be. Anyways, it's hardly useful generally, AFAIK we only use it to filter locations _if_ `preferred_layer` is used.

I never had a graph build crash though, did you?

**Comment by mkandulavm** [comment]
>I never had a graph build crash though, did you?
Yes. When processing the OSM for southern zone of India, the layer value provided was not an int.

> it's hardly useful generally
I think its useful when render layers are needed to enable/disable the view of them. For example, the metro train line seems to be using this. 

I have to check, but if you require , I can recollect and provide the coordinate where its used and where the value is not an int.

**Comment by nilsnolde** [comment]
> When processing the OSM for southern zone of India, the layer value provided was not an int.

That must've been an unfortunate state of OSM and must've been reverted immediately. I'm building the planet for the public instance on a 1.5 daily basis, didn't crash in more than a year bcs of OSM.

Ah sure, `layer` is very crucial for map rendering, was talking in the context of routing/Valhalla.

No need to give an example. Let's add it to c++ instead of lua, more transparent IMO.
